File: P1/E3/Jefe.java
package E3;
import java.util.ArrayList;

public class Jefe implements Empleado{

    private String nombre;
    ArrayList<Empleado> empleados = new ArrayList<Empleado>(); //LISTA DE EMPLEADOS A SU CARGO.
    
    public Jefe(String new_nombre){
        this.nombre = new_nombre;
    }

    public void aniadeEmpleado(Empleado empleado){
        empleados.add(empleado);
    }

    public void eliminaEmpleado(Empleado empleado){
        empleados.remove(empleado);
    }

    @Override
    public String getNombre(){
        return ("Jefe - " + nombre);
    }

    public ArrayList<Empleado> getEquipo(){
        return empleados;
    }
    
}

File: P2/lib/jefe.dart
import 'dart:core';
import 'empleado.dart';

class Jefe implements Empleado {
  late String nombre;
  final List<Empleado> _empleados = [];

  Jefe(String nomb) {
    nombre = nomb;
  }

  void aniadeEmpleado(Empleado empleado) {
    _empleados.add(empleado);
  }

  void eliminaEmpleado(Empleado empleado) {
    _empleados.remove(empleado);
  }

  @override
  String getNombre() {
    return "Jefe - $nombre";
  }

  List<Empleado> getEquipo() {
    return _empleados;
  }
}
